terraform/eval_diff.go

```diff
@@ -495,6 +482,12 @@ func (n *EvalDiff) Eval(ctx EvalContext) (interface{}, error) {
 		priorVal = priorValTainted
 	}
 
+	// If we plan to write or delete sensitive paths from state,
+	// this is an Update action
+	if action == plans.NoOp && len(priorPaths) != len(unmarkedPaths) {
```

Comment:
Is it possible for two attributes to change sensitivity at the same time, so that `len(priorPaths) == len(unmarkedPaths)`, but the actual marked path set has changed?

Reply:
Ah very true! This could be possible.